discordia/client/API.lua

```diff
@@ -72,7 +72,12 @@ function API:request(method, route, endpoint, payload)
 	end
 
 	if method:find('P') then
-		payload = payload and encode(payload) or '{}'
+		if isMulti then
+			reqHeaders[3][2] = 'multipart/form-data; boundary=Discordia'
```

Comment:
It's not for two reasons. The iterator order of `pairs` cannot be trusted and self._headers` may not always been the same. The appropriate header needs to be explicitly located. You can't assume that it's always the third one.